shared/common-adapters/form-input.native.js

```diff
@@ -0,0 +1,165 @@
+  state: State
+  _input: any
+
+  constructor(props: Props) {
+    super(props)
+
+    this.state = {
+      value: this.props.value || '',
+      focused: false,
+    }
+  }
+
+  getValue = (): string => {
+    return this.state.value || ''
+  }
+
+  setValue = (value: string) => {
+    this.setState({value: value || ''})
+  }
+
+  clearValue = () => {
+    this._onChangeText('')
+  }
+
+  focus = () => {
+    this._input && this._input.focus()
+  }
+
+  blur = () => {
+    this._input && this._input.blur()
+  }
+
+  select = () => {
+    this._input && this._input.select()
+  }
+
+  _onChangeText = (text: string) => {
+    this.setState({value: text || ''})
+
+    this.props.onChangeText && this.props.onChangeText(text || '')
+  }
+
+  _onFocus = () => {
+    this.setState({
+      focused: true,
+    })
+  }
+
+  _onBlur = () => {
+    this.setState({
+      focused: false,
+    })
+  }
+
+  _onKeyPress = (e: SyntheticKeyboardEvent<>) => {
+    if (this.props.onEnterKeyDown && e.key === 'Enter') {
+      this.props.onEnterKeyDown()
+    }
+  }
+
+  render() {
+    let backgroundColor = this.props.backgroundMode
+      ? backgroundModeToColor[this.props.backgroundMode]
+      : globalColors.white
+    return (
+      <Box
+        style={{
+          ...globalStyles.flexBoxColumn,
+          alignItems: 'center',
+          position: 'relative',
+          width: '100%',
+          borderTopWidth: this.props.hideTopBorder ? 0 : 1,
+          borderBottomWidth: this.props.hideBottomBorder ? 0 : 1,
+          borderColor: globalColors.black_10,
+          minHeight: this.props.multiline ? 128 : 64,
+          ...(this.props.maxHeight ? {maxHeight: this.props.maxHeight} : {}),
+          ...this.props.containerStyle,
+        }}
+      >
+        {(!!this.state.value || this.state.focused) && (
+          <Text
+            type="BodySmallSemibold"
+            style={{
+              position: 'absolute',
+              top: globalMargins.tiny,
+              left: globalMargins.small,
+              color: globalColors.blue,
+              fontSize: 11,
+              zIndex: 2,
+              backgroundColor: globalColors.transparent,
+            }}
+          >
+            {this.props.label}
+          </Text>
+        )}
+        {this.props.label &&
+          this.props.multiline && (
+            <Box
+              style={{
+                position: 'absolute',
+                top: 0,
+                left: 0,
+                right: 0,
+                height: 22,
+                backgroundColor: backgroundColor,
+                zIndex: 1,
+              }}
+            />
+          )}
+        <NativeTextInput
+          autoCorrect={this.props.autoCorrect}
+          autoFocus={this.props.autoFocus}
+          autoGrow={!!this.props.multiline && !!this.props.maxHeight}
+          value={this.state.value}
+          onChangeText={this._onChangeText}
+          placeholder={this.state.focused ? '' : this.props.label}
```

Comment:
There's a bug here in multiline inputs where if the input is touched but unfocused when empty, the placeholder can disappear. I spent some time working on it but couldn't make any progress